src/docfx/build/page/BuildPage.cs

```diff
@@ -114,19 +114,29 @@ public static async Task<List<Error>> Build(Context context, Document file)
 
             if (context.Config.Output.Json && !context.Config.Legacy)
             {
-                return (errors, outputModel, SortProperties(outputMetadata));
+                return (errors, outputModel, (JObject)SortProperties(outputMetadata));
             }
 
-            var (templateModel, templateMetadata) = CreateTemplateModel(context, SortProperties(outputModel), file);
+            var (templateModel, templateMetadata) = CreateTemplateModel(context, (JObject)SortProperties(outputModel), file);
             if (context.Config.Output.Json)
             {
-                return (errors, templateModel, SortProperties(templateMetadata));
+                return (errors, templateModel, (JObject)SortProperties(templateMetadata));
             }
 
             var html = context.TemplateEngine.RunLiquid(file, templateModel);
-            return (errors, html, SortProperties(templateMetadata));
+            return (errors, html, (JObject)SortProperties(templateMetadata));
 
-            JObject SortProperties(JObject obj) => new JObject(obj.Properties().OrderBy(p => p.Name));
+            JToken SortProperties(JToken jToken)
```

Comment:
nit: consider move it to `JsonUtility.cs` and keep the existing method signature to avoid the `(JObject)` hard cast. You can check if `property.Value` is `JObject` before calling `SortProperties` in line 136.